lib/seahorn/BmcPass.cc

```diff
@@ -348,24 +348,24 @@ class BmcPass : public llvm::ModulePass {
           errs() << "CORE END\n";
         });
 
-    LOG(
-        "cex", if (res) {
-          errs() << "Analyzed Function:\n" << F << "\n";
-          errs() << "Trace \n";
-          BmcTrace trace(bmc.getTrace());
-          trace.print(errs());
-        });
-
     if (res) {
+      ZBmcTraceTy trace(bmc.getTrace());
+      LOG(
+          "cex", if (res) {
+            errs() << "Analyzed Function:\n" << F << "\n";
+            errs() << "Trace \n";
+            trace.print(errs());
+          });
+
+      // dump to harness file
       StringRef CexFileRef(HornCexFile);
       if (CexFileRef != "") {
         if (CexFileRef.endswith(".ll") || CexFileRef.endswith(".bc")) {
           auto &tli = getAnalysis<TargetLibraryInfoWrapperPass>();
           auto const &dl = F.getParent()->getDataLayout();
-          BmcTrace trace(bmc.getTrace());
-          BmcTraceWrapper trace_wrapper(trace);
-          dumpLLVMCex(trace_wrapper, CexFileRef, dl, tli.getTLI(F),
-                      F.getContext());
+          BmcTraceWrapper<ZBmcTraceTy> trace_wrapper(trace);
+          dumpLLVMCex<ZBmcTraceTy>(trace_wrapper, CexFileRef, dl, tli.getTLI(F),
```

Comment:
`ZBmcTraceTy` should not be needed. If it is needed, then something is wrong with how it is declared.